packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -219,10 +216,15 @@ export class ProfilesCache {
                         // Step 3: Update allProfiles list
                         tmpAllProfiles.push(profileFix);
                     }
-                    this.allProfiles.push(...tmpAllProfiles);
+                    allProfiles.push(...tmpAllProfiles);
                     this.profilesByType.set(type, tmpAllProfiles);
                 }
-                this.allTypes.push(type);
+            }
+            this.allProfiles = allProfiles;
+            this.allTypes = allTypes;
+            for (const oldType of [...this.profilesByType.keys()].filter((type) => !allTypes.includes(type))) {
+                this.profilesByType.delete(oldType);
+                this.defaultProfileByType.delete(oldType);
```

Comment:
I know this for-loop is based on a filtered version of `this.profileByType`, so doing `this.profileByType.delete(oldType)` should be fine.
I'm curious if we should check if `this.defaultProfileByType.has(oldType)` before trying to delete it. 😋

Reply:
The current logic should be okay, as `delete` will [return false](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete#return_value) if the element doesn't exist